src/wallet/mod.rs

```diff
@@ -256,10 +256,12 @@ where
 
     // Return the the last previously derived address if it has not been used in a received
     // transaction. Otherwise return a new address using [`Wallet::get_new_address`].
-    fn get_unused_address(&self) -> Result<AddressInfo, Error> {
-        let current_index = self.fetch_index(KeychainKind::External)?;
+    fn get_unused_address(&self, keychain: KeychainKind) -> Result<AddressInfo, Error> {
```

Comment:
Though this one doesn't mention `external` explicitly, maybe better to modify too for sake of completeness?